### DISCUSSION TITLE
The total charge of the smiles string is not being computed correctly

### DISCUSSION THREAD

**Discussion by xiki-tempula** [rdkit/rdkit#4331] [Q&A] [answered]
I'm interested in using rdkit to obtain the total charge of the smiles string for my QM calculations. 
My pipeline is that I will add hydrogen atoms, calculate the Gasteiger Charges and the sum of Gasteiger Charges is the total charge of the molecules.

```python
mol = Chem.MolFromSmiles(string)
mol_H = Chem.AddHs(mol)
AllChem.ComputeGasteigerCharges(mol_H)
contribs = [mol_H.GetAtomWithIdx(i).GetDoubleProp('_GasteigerCharge')
                    for i in range(mol_H.GetNumAtoms())]
charge = int(sum(contribs))
```

However, this pipeline won't work for the string 'c1cc(oc1)CN=C2C=C(C(=CC2=C(O)[O-])S(=O)(=O)[NH-])Cl', which clearly has total charge of -2, but the above pipeline will give a total charge of -1.

the rdkit version is '2021.03.4'.

**Comment by greglandrum** [comment, marked as answer]
Hi @xiki-tempula,
You should use `round()` instead of `int()` in Python:
```
In [13]: m = Chem.AddHs(Chem.MolFromSmiles('c1cc(oc1)CN=C2C=C(C(=CC2=C(O)[O-])S(=O)(=O)[NH-])Cl'))

In [14]: rdPartialCharges.ComputeGasteigerCharges(m)

In [15]: contribs = [x.GetDoubleProp('_GasteigerCharge') for x in m.GetAtoms()]

In [16]: sum(contribs)
Out[16]: -1.9999999999999996

In [18]: round(sum(contribs))
Out[18]: -2
```
Python's `int()` function doesn't round to the nearest integer; it gives you the next lowest integer.

**Reply by xiki-tempula** [reply]
I see. Thank you.
